tests/common.bash

```diff
@@ -1,14 +1,62 @@
 #!/bin/bash
 
-function clean_up_environment {
-	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found pods --all
-	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found -n kubewarden clusteradmissionpolicies --all
-	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found -n kubewarden admissionpolicies --all
-	kubectl --context $CLUSTER_CONTEXT wait --for=condition=Ready -n kubewarden pod --all
+bats_require_minimum_version 1.5.0
+
+function kubectl() {
+	command kubectl --context $CLUSTER_CONTEXT "$@"
+}
+
+function helm() {
+	command helm --kube-context $CLUSTER_CONTEXT "$@"
+}
+
+function retry() {
+    local cmd=$1
+    local tries=${2:-10}
+    local delay=${3:-30}
+    local i
+    for ((i=1; i<=tries; i++)); do
+        timeout 25 bash -c "$cmd" && break || echo "RETRY #$i: $cmd"
+        [ $i -ne $tries ] && sleep $delay || { echo "Godot: $cmd"; false; }
+    done
+}
+
+# Safe version of vaiting for pods
+# Handles kube-api disconnects during upgrade
+function wait_all_pods() {
+    local i output
+    for i in {1..90}; do
+        output=$(kubectl get pods --no-headers -n kube-system -o wide | grep -vw Completed || echo 'Fail')
```

Comment:
should we wait for pods in all namespaces here?

Reply:
yes, I stole this from caasp :) - it's reliable way to check pods/cluster are up & ready to handle next task.
You can use parameter to wait for something specific.
```bash
# reliable
wait_pods # calls wait_all_pods which waits for all pods in all namespaces

# less reliable
wait_pods -n kubewarden # wait for something specific, any param for kubectl wait can be used
```

Reply:
ah, I read it again and see now what you meant, will look into it tomorrow